Paper-MojangAPI/src/main/java/io/papermc/paper/brigadier/PaperBrigadierProvider.java

```diff
@@ -0,0 +1,30 @@
+package io.papermc.paper.brigadier;
+
+import com.mojang.brigadier.Message;
+import net.kyori.adventure.text.Component;
+import net.kyori.adventure.text.ComponentLike;
+import org.checkerframework.checker.nullness.qual.NonNull;
+import org.checkerframework.checker.nullness.qual.Nullable;
+
+import static java.util.Objects.requireNonNull;
+
+interface PaperBrigadierProvider {
+    class Holder {
+        private static PaperBrigadierProvider INSTANCE;
+    }
+
+    static @NonNull PaperBrigadierProvider instance() {
+        return requireNonNull(Holder.INSTANCE, "PaperBrigadierProvider has not yet been initialized!");
+    }
+
+    static void initialize(final @Nullable PaperBrigadierProvider instance) {
```

Comment:
Document this should only be initialised by the server. `@Deprecated` would be a completely okay annotation here.

Reply:
I mean, the class and method are package private, so doesn't matter that much

Reply:
Didn't even notice 😅